src/app/hooks/useLocalDismissal.ts

```diff
@@ -0,0 +1,84 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import { useState } from "react";
+import { useCookies } from "react-cookie";
+
+export type DismissOptions = {
+  /** If true, the dismissal won't take effect right away, but the cookie to store the dismissal _will_ be set. */
+  soft?: boolean;
+};
+
+export type DismissalData = {
+  isDismissed: boolean;
+  dismiss: (options?: DismissOptions) => void;
+};
+
+export type DismissalOptions = {
+  /** How long the dismissal should last, in seconds. Note that by default, the cookie will be deleted in 30 days. */
+  duration?: number;
+};
+
+/**
+ * This can be used to store a cookie to remember that something was dismissed.
+ *
+ * @param key Key to identity the item-to-be-dismissed with (i.e. to use in the cookie name). Tip: incorporate the user's ID to make a dismissal user-specific.
+ * @param options See {@link DismissalOptions}.
+ * @returns Whether the item has been dismissed yet, and a function to call to dismiss the item.
+ */
+export function useLocalDismissal(
+  key: string,
+  options: DismissalOptions = {},
+): DismissalData {
+  const cookieId = key + "_dismissed";
+  const [cookies, setCookie] = useCookies([cookieId]);
+  const cookieValue = cookies[cookieId] as string | undefined;
+
+  const [isDismissed, setIsDismissed] = useState(
+    hasDismissedCookie(cookieValue, options.duration),
+  );
+
+  const dismiss = (dismissOptions?: DismissOptions) => {
+    const maxAgeInSeconds =
+      typeof options.duration === "number"
+        ? options.duration
+        : 100 * 365 * 24 * 60 * 60;
```

Comment:
100 years?

Reply:
"forever", basically. (There were a couple of instances on Relay where we show something again if it was dismissed more than `x` time ago. Maybe I should just remove it here until we find we actually need it.)